src/Tools/ExternalAccess/FSharp/PublicAPI.Unshipped.txt

```diff
@@ -0,0 +1,181 @@
+Microsoft.CodeAnalysis.ExternalAccess.FSharp.Classification.ClassificationTags
+Microsoft.CodeAnalysis.ExternalAccess.FSharp.Classification.IFSharpClassificationService
```

Comment:
This file should be empty. We can't be exposing new public APIs. The ExternalAccess.FSharp should keep IVT to FSharp

Reply:
@tmat is correct. All top-level types in this assembly should be `internal`.

Reply:
I'm a little confused. I'm making public shim types that don't expose internals to Roslyn, that is specific for consumption in F#. If I make all types `internal`, that means F# has IVTs to internal types that I don't want exposed or to be used.

Reply:
We are fine exposing internals of the ExternalAccess.FSharp assembly to FSharp, but not any other Roslyn assembly.

Reply:
This is a trade-off. We decided in favor of `internal` to ensure that other extension developers don't try to use the various ExternalAccess assemblies as a way to access Roslyn functionality that doesn't have public APIs.

Reply:
I need to think about this. The only developer who is going to use this is VisualFSharp. I don't feel comfortable making everything internal and giving IVT to FSharp.Editor. What if I have internal types that use actual Roslyn internal types? Or types that I deemed truly internal for use only inside ExternalAccess. In FSharp.Editor I would see them; that scares me.

I see your point. You don't want developers to try to use our internal services, even though the real internal types are not exposed. The way I see it, if a developer tries to use `ExternalAccess.*` as a means to gain access to functionality, they made a poor choice and will ultimately be broken.

Reply:
>  The way I see it, if a developer tries to use ExternalAccess.* as a means to gain access to functionality, they made a poor choice and will ultimately be broken.

That may very well be but the expectation is that any public API is a contract to not be broken. So we can't make these public.

Reply:
> What if I have internal types that use actual Roslyn internal types?

You will not be able to access the Roslyn internal types from F#. I can see how it might be an annoyance though - when you're writing the shim you'd have no indication that you are exposing an internal Roslyn type. You'll only find out when consuming the shim in F#. An analyzer might be useful here.

Reply:
I understand, I still feel a little weird about it. What I'm going to do is put those types that I deem not to be used on the VisualFSharp side in a different namespace.

I'll make everything else internal.

Reply:
Ok, it's done.

Reply:
> when you're writing the shim you'd have no indication that you are exposing an internal Roslyn type

I thought this would be annoying, but then I realized the local testing process for someone creating external access APIs would involve building the Roslyn side of things and then compiling against it, so while you can't tell from a Roslyn build in isolation that an API is unusable, you would immediately find out during the first round of local testing. I'm hoping it's not problematic in the end.